### PR TITLE
Updated RigidBodyTree::findAncestorBodies() to be style guide compliant.

### PR DESCRIPTION
**Pull request by liangfok**
Renamed `RigidBodyTree::findAncestorBodies()` to be `RigidBodyTree::FindAncestorBodies()`. Updated parameter order and types to be style guide compliant.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/3168)

### CODE REVIEW

**Comment by liangfok** [comment]
Since this PR touches `RigidBody*`:

+@amcastro-tri for feature review.
+@sherm1 for platform review.

Thanks!

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPLHCrzK08bO0VdEjLH:b-4rplao)*

**Comment by hongkai-dai** [comment]
:lgtm:

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/plants/RigidBodyTree.h, line 267 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPLMxX7g-JYR8-D0Mf6:-KPLMxX8LBJ7pHQeRWJM:bfdr64k) ([raw file](https://github.com/robotlocomotion/drake/blob/9be6072d31f9853fa7a861db9520566cc2dfb81f/drake/systems/plants/RigidBodyTree.h#L267)):*
> ```objc
>    * indices are stored in @p ancesor_bodies.
>    *
>    * @param[out] ancestor_bodies A pointer to where the indices of the ancestor
> ```

It was not a pointer, but a reference in the previous API.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPLN3kKo5Rb-KAfQZF7:bnfp4nl)*

**Comment by hongkai-dai** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/plants/RigidBodyTree.h, line 267 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPLMxX7g-JYR8-D0Mf6:-KPLN8ggh-u7LngeiB0H:bpyblyc) ([raw file](https://github.com/robotlocomotion/drake/blob/9be6072d31f9853fa7a861db9520566cc2dfb81f/drake/systems/plants/RigidBodyTree.h#L267)):*
<details><summary><i>Previously, hongkai-dai (Hongkai Dai) wrote…</i></summary>
> It was not a pointer, but a reference in the previous API.
</details>
Sorry, just realized this documentation is for the new API. Ignore my previous comment.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168)*

**Comment by liangfok** [comment]
Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---

*[drake/systems/plants/RigidBodyTree.h, line 267 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPLMxX7g-JYR8-D0Mf6:-KPLVf2VyccT-Qvpj02q:b-vqkjuf) ([raw file](https://github.com/robotlocomotion/drake/blob/9be6072d31f9853fa7a861db9520566cc2dfb81f/drake/systems/plants/RigidBodyTree.h#L267)):*
<details><summary><i>Previously, hongkai-dai (Hongkai Dai) wrote…</i></summary>
> Sorry, just realized this documentation is for the new API. Ignore my previous comment.
</details>
No worries.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168)*

**Comment by sherm1** [comment]
Reviewed -- PTAL.

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/systems/plants/RigidBodyTree.h, line 270 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPLxpEfHq4X512h4ZlI:-KPLxpEfHq4X512h4ZlJ:bshxyk4) ([raw file](https://github.com/robotlocomotion/drake/blob/9be6072d31f9853fa7a861db9520566cc2dfb81f/drake/systems/plants/RigidBodyTree.h#L270)):*
> ```objc
>    * bodies should be stored.
>    */
>   void FindAncestorBodies(int body_index, std::vector<int>* ancestor_bodies)
> ```

@liangfok, since you are changing the signature anyway why not make it:
```c++
std::vector<int> RigidBodyTree::FindAncestorBodies(int body_index) const;
```
and take advantage of return value optimization to make a nicer API?

Also, whatever signature is used here should dictate the signatures for any new methods that come out of issue #3142.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPLxu5I9eezzxIU-Jhc:b-9lf7dp)*

**Comment by liangfok** [comment]
I addressed all reviewer comments. PTAL. Thanks.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/plants/RigidBodyTree.h, line 270 \[r1\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPLxpEfHq4X512h4ZlI:-KPNvNj1OJ1n9ePHW7YU:bjmlu) ([raw file](https://github.com/robotlocomotion/drake/blob/9be6072d31f9853fa7a861db9520566cc2dfb81f/drake/systems/plants/RigidBodyTree.h#L270)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> @liangfok, since you are changing the signature anyway why not make it:
> ```c++
> std::vector<int> RigidBodyTree::FindAncestorBodies(int body_index) const;
> ```
> and take advantage of return value optimization to make a nicer API?
> 
> Also, whatever signature is used here should dictate the signatures for any new methods that come out of issue #3142.
</details>
I modified the API as you suggest. It is indeed much cleaner now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPNvpWLjfOOQmmnZAk3:bblnx7v)*

**Comment by sherm1** [comment]
Platform: :lgtm: -- a few BTWs.

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[drake/systems/plants/RigidBodyTree.h, line 267 \[r2\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPPBJZlzJqyqCkJJ-NR:-KPPBJZlzJqyqCkJJ-NS:b-9tr1zq) ([raw file](https://github.com/robotlocomotion/drake/blob/983d7145f8e640e75e1c2c943666e25c20fc41f7/drake/systems/plants/RigidBodyTree.h#L267)):*
> ```objc
>    * indices are stored in @p ancesor_bodies.
>    *
>    * @return A vector of indices of the ancestor bodies of the body at index
> ```

BTW, I was about to write "indices" somewhere but I looked up "index" [here](http://www.dictionary.com/browse/index) and discovered that "indexes" is also allowed and actually appears to be preferred (since it comes first). I am planning to retrain myself to use that since "indices" seems less obviously related to "index".

---

*[drake/systems/plants/RigidBodyTree.h, line 270 \[r2\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPPClvC1U8clxPVumSn:-KPPClvC1U8clxPVumSo:b-34xs73) ([raw file](https://github.com/robotlocomotion/drake/blob/983d7145f8e640e75e1c2c943666e25c20fc41f7/drake/systems/plants/RigidBodyTree.h#L270)):*
> ```objc
>    * @p body_index.
>    */
>   std::vector<int> FindAncestorBodies(int body_index) const;
> ```

BTW, looks great this way!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPPCs3mx_mLZmnsDfBz:b9jf60f)*

**Comment by liangfok** [comment]
Just waiting for CI to pass now.

---

Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/systems/plants/RigidBodyTree.h, line 267 \[r2\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPPBJZlzJqyqCkJJ-NR:-KPPE94YqYbkugsgsLlk:b-itqlup) ([raw file](https://github.com/robotlocomotion/drake/blob/983d7145f8e640e75e1c2c943666e25c20fc41f7/drake/systems/plants/RigidBodyTree.h#L267)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> BTW, I was about to write "indices" somewhere but I looked up "index" [here](http://www.dictionary.com/browse/index) and discovered that "indexes" is also allowed and actually appears to be preferred (since it comes first). I am planning to retrain myself to use that since "indices" seems less obviously related to "index".
</details>
Interesting. TIL. I'll also switch to use "indexes" for the same reason.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPPETqs1ag0RufQJHS6:b-t704pp)*

**Comment by sherm1** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168)*

**Comment by liangfok** [comment]
The latest round of CI showed a number of unit tests failing:

https://drake-cdash.csail.mit.edu/viewTest.php?onlyfailed&buildid=82900

I have so far been unable to reproduce the failures locally.

---

Review status: 2 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPSZy3Hk5mLIuop2xpO:bwxroi)*

**Comment by amcastro-tri** [comment]
Reviewed 1 of 1 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[CHANGELOG.md, line 26 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPSu1Bj62FCw0kJTMtT:-KPSu1Bj62FCw0kJTMtU:b-4qh37v) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/CHANGELOG.md#L26)):*
> Modified method to return a vector of indices rather than add to an output vector parameter.

Though I know what you meant (I read the PR) it sounds a bit strange to me

---

*[drake/systems/plants/RigidBodyTree.h, line 265 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPSsKx9obTMAVSSkAx8:-KPSsKx9obTMAVSSkAx9:b-b0jdp1) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/drake/systems/plants/RigidBodyTree.h#L265)):*
> ```objc
>    * @param[in] body_index An index within this RigidBodyTree's vector of
>    * bodies. The ancestors of the body at this index are found and their
>    * indexes are stored in @p ancesor_bodies.
> ```

The user does not need to know that the RBT stores a vector of bodies (it could be a list or something else, it doesn't matter). Therefore I would rephrase this to:
`
@param[in] body_index The index of the body in this RigidBodyTree for which the ancestors of the body are found. Ancestor's are returned in a vector of body indexes.
`

---

*[drake/systems/plants/RigidBodyTree.h, line 265 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPStHEywOf9xYCBRCHi:-KPStHEywOf9xYCBRCHj:btnx149) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/drake/systems/plants/RigidBodyTree.h#L265)):*
> ```objc
>    * @param[in] body_index An index within this RigidBodyTree's vector of
>    * bodies. The ancestors of the body at this index are found and their
>    * indexes are stored in @p ancesor_bodies.
> ```

`@p ancesor_bodies`:
1. Typo
2. Is not a parameter.

---

*[drake/systems/plants/RigidBodyTree.h, line 267 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPSsEeYJDJ9Vxcm4NOA:-KPSsEeYJDJ9Vxcm4NOB:b-j28s44) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/drake/systems/plants/RigidBodyTree.h#L267)):*
> ```objc
>    * indexes are stored in @p ancesor_bodies.
>    *
>    * @return A vector of indexes of the ancestor bodies of the body at index
> ```

`...of the body at index...` -->`...of the body with index...`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168)*

**Comment by liangfok** [comment]
I addressed all reviewer comments. PTAL.

---

Review status: 1 of 3 files reviewed at latest revision, 4 unresolved discussions.

---

*[CHANGELOG.md, line 26 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPSu1Bj62FCw0kJTMtT:-KPT2WP1kvVR62U1hkJd:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/CHANGELOG.md#L26)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> > Modified method to return a vector of indices rather than add to an output vector parameter.
> 
> Though I know what you meant (I read the PR) it sounds a bit strange to me
</details>
Done.

---

*[drake/systems/plants/RigidBodyTree.h, line 265 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPSsKx9obTMAVSSkAx8:-KPT2ohKhNtx8GbIy2HK:b-rsmgn0) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/drake/systems/plants/RigidBodyTree.h#L265)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> The user does not need to know that the RBT stores a vector of bodies (it could be a list or something else, it doesn't matter). Therefore I would rephrase this to:
> `
> @param[in] body_index The index of the body in this RigidBodyTree for which the ancestors of the body are found. Ancestor's are returned in a vector of body indexes.
> `
> 
</details>
Done.> @param[in] body_index The index of the body in this RigidBodyTree for which the ancestors of the body are found. Ancestor's are returned in a vector of body indexes.

---

*[drake/systems/plants/RigidBodyTree.h, line 265 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPStHEywOf9xYCBRCHi:-KPT2uBQ2i2nbgO2Fk_Z:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/drake/systems/plants/RigidBodyTree.h#L265)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> `@p ancesor_bodies`:
> 1. Typo
> 2. Is not a parameter.
</details>
Done.

---

*[drake/systems/plants/RigidBodyTree.h, line 267 \[r4\]](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-KPSsEeYJDJ9Vxcm4NOA:-KPT3INLSXu8DAdkwNq4:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/19fde8136578e3808acf338336e320268383c82b/drake/systems/plants/RigidBodyTree.h#L267)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary>
> `...of the body at index...` -->`...of the body with index...`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPT3bJrNf8CPQBrRD1Q:b-1edt96)*

**Comment by amcastro-tri** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPT4DxBvQlBFsqRAxcA:bnfp4nl)*

**Comment by liangfok** [comment]
Thanks. Now waiting on CI.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3168#-:-KPT4MssUObqFdhDlbQW:b-mta4n0)*
